Allow setting a custom backend storage

esolitos opened the pull request:
With this change the secondary storage bin can be customised on demand, allowing to define, for example, another memory storage.
This is  can be particularly useful in conjunction with wodby/drupal-varnish#3 because it allows the bin to be completely customised.

_Note_ This is an extension of `VARNISHD_STORAGE_SIZE`, making it more customisable while preserving backward compatibility.

csandanov commented:
I think it would be better to use something like `VARNISHD_SECONDARY_STORAGE` where you can specify storage name, type, path and size.

esolitos commented:
I agree, that `VARNISHD_SECONDARY_STORAGE` is a better name for the variable _(changed it now)_, however I was thinking to leave the storage name fixed to `secondary`, just to prevent the need for a second variable for the storage name in the use-case: wodby/drupal-varnish#3
However if you think it's better leaving the storage name configurable as well, I'll change that as well.

esolitos commented:
Ok, the idea was to leave `VARNISHD_STORAGE_SIZE` for backward compatibility, do you think I should simply drop it?

Last commits are just small changes: renamed the variable and changed the priority since the `VARNISHD_STORAGE_SIZE` would be deprecated.

esolitos commented:
Updated dropping `VARNISHD_STORAGE_SIZE` in favour of the more generic `VARNISHD_SECONDARY_STORAGE`.
